lib/dynflow/executors/sidekiq/worker_jobs.rb

```diff
@@ -0,0 +1,34 @@
+module Dynflow
+  module Executors
+    module Sidekiq
+      module WorkerJobs
+        class PerformWork < InternalJobBase
+          def perform(work_item)
+            with_telemetry(work_item) do
+              Executors.run_user_code do
+                work_item.world = Dynflow.process_world
+                work_item.execute
+              end
+            end
+          rescue Errors::PersistenceError => e
+            OrchestratorJobs::HandlePersistenceError.perform_async(e, work_item)
+          ensure
+            OrchestratorJobs::WorkerDone.perform_async(work_item)
```

Comment:
When a persistence error happens, both jobs get queued. However on orchestrator's side, the first one which arrives (`HandlePersistenceError`) removes the execution plan from all the internal structures and then the `WorkerDone` is just a no-op. Do we really need to send both?

When an `Errors::FatalPersistenceError` is raised on the worker, it gets caught and sent back to the orchestrator. The orchestrator reacts to this by terminating. There are two things wrong with this: the worker which had the error keeps running and the orchestrator which may be working fine terminates.

Reply:
And what's even worse is the orchestrator process does not die, just the world inside it terminates